java/alizer-api/src/main/java/com/redhat/devtools/alizer/api/ServiceRecognizerImpl.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.alizer.api;
+
+import com.redhat.devtools.alizer.api.spi.framework.FrameworkDetectorProvider;
+import com.redhat.devtools.alizer.api.spi.framework.FrameworkDetectorWithConfigFileProvider;
+import com.redhat.devtools.alizer.api.spi.framework.FrameworkDetectorWithoutConfigFileProvider;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+public class ServiceRecognizerImpl extends Recognizer {
+    public ServiceRecognizerImpl(RecognizerFactory builder) {
+        super(builder);
+    }
+
+    public Map<Path, List<Service>> analyze(String path) throws IOException {
+        ComponentRecognizer componentRecognizer = new RecognizerFactory().createComponentRecognizer();
+        List<Component> components = componentRecognizer.analyze(path);
+
+        Map<Path, List<Service>> services = new HashMap<>();
+        for(Component component: components) {
+            List<Service> componentServices = getServices(component);
```

Comment:
Seems services are guessed per component but what if you have a repo that has 2 components in 2 sub directories and a docker-compose.yml at the root that may start the 2 components but also other common services. Or maybe that's a feature for the future as I'm not sure odo allows it but we may maybe have a more general design and implementation